### PR TITLE
Increase probability of synthetic source tests using ignore_malformed

### PR DESCRIPTION
**Pull request by lkts**
In scope of ongoing work on `ignore_malformed` support in synthetic source (#106483) i noticed that `rarely()` used in synthetic source tests to determine application of `ignore_malformed` is indeed very rare. Based on my testing it has a probability significantly lower than 1%. Given that a lot of fields now have support for malformed values in synthetic source it seems reasonable to exercise this code more often. This PR changes the logic to apply `ignore_malformed` in 5% of tests.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java [hunk 1]
```diff
@@ -1268,6 +1267,11 @@ public final void testSyntheticEmptyList() throws IOException {
         assertThat(syntheticSource(mapper, b -> b.startArray("field").endArray()), equalTo(expected));
     }
 
+    private boolean shouldUseIgnoreMalformed() {
+        // 5% of test runs use ignore_malformed
+        return supportsIgnoreMalformed() && randomDouble() >= 0.05;
```
Comment (dnhatn) [inline test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java:1272]: should be `<=` instead?

**Review by dnhatn — APPROVED** [review]
One comment. Feel free to merge after addressing it. Thanks @lkts.

File: test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java (hunk 1 above)
Comment (lkts) [reply test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java:1272]: Wow, of course. I guess it's a good stress test to run this on CI :)

**Comment by lkts** [comment]
@elasticmachine update branch

**Comment by lkts** [comment]
@elasticmachine update branch

**Pull request merged**